### ISSUE TITLE
Error: Cannot read property 'headers' of undefined in Client.js when offline

### ISSUE THREAD

**Issue by WolfWalter** [aws-amplify/amplify-js#389]
**Do you want to request a feature or report a bug?**
bug

**What is the current behavior?**
When CognitoUser refreshSession gets called while being offline, Client.js on line 67 tries to use an empty response object resulting in an uncaught error:

```
ERROR Error: Uncaught (in promise): TypeError: Cannot read property 'headers' of undefined
TypeError: Cannot read property 'headers' of undefined
    at Client.js:67
    at t.invoke (polyfills.js:3)
    at Object.onInvoke (core.js:4630)
    at t.invoke (polyfills.js:3)
    at r.run (polyfills.js:3)
    at polyfills.js:3
    at t.invokeTask (polyfills.js:3)
    at Object.onInvokeTask (core.js:4621)
    at t.invokeTask (polyfills.js:3)
    at r.runTask (polyfills.js:3)
    at Client.js:67
    at t.invoke (polyfills.js:3)
    at Object.onInvoke (core.js:4630)
    at t.invoke (polyfills.js:3)
    at r.run (polyfills.js:3)
    at polyfills.js:3
    at t.invokeTask (polyfills.js:3)
    at Object.onInvokeTask (core.js:4621)
    at t.invokeTask (polyfills.js:3)
    at r.runTask (polyfills.js:3)
    at c (polyfills.js:3)
    at polyfills.js:3
    at t.invokeTask (polyfills.js:3)
    at Object.onInvokeTask (core.js:4621)
    at t.invokeTask (polyfills.js:3)
    at r.runTask (polyfills.js:3)
    at o (polyfills.js:3)
    at <anonymous>
```

**What is the expected behavior?**
Returning null or throwing an error / rejecting.

**Which versions of Amplify, and which browser / OS are affected by this issue? Did this work in previous versions?**
aws-amplify 0.2.9 
Linux (Ubuntu 16.04) in Chrome 64 and Electron 1.8.2

**Comment by davideickhoff** [comment]
the issue is quite obvious and easy to see in the code (Client.js)
```
59        // Taken from aws-sdk-js/lib/protocol/json.js
60        const code = (response.headers.get('x-amzn-errortype') || 'UnknownError').split(':')[0];
```

the response object is defined with `let response` and if the fetch call fails, there is no assignment to the response object.

In one of our applications I have customer reports (sent through sentry.io) which describe the same error. I am not sure if it is related to an offline issue (their browser was able to send me the sentry.io report, so they should have been online somehow). It may occur when switching networks or trying to refresh tokens.

**Comment by ArmorDarks** [comment]
Just stumbled into the same issue. The most disappointing part is that right now there is no way to catch that error.

**Comment by maro1987** [comment]
I solved it by downgrading `amazon-cognito-identity-js` to version `1.29.0`

**Comment by elorzafe** [comment]
This issue was fixed on this PR #605
